src/librepgp/stream-ctx.cpp

```diff
@@ -74,19 +75,16 @@ rnp_ctx_add_encryption_password(rnp_ctx_t *    ctx,
     if (!pgp_s2k_derive_key(&info.s2k, password, info.key, sizeof(info.key))) {
         return RNP_ERROR_GENERIC;
     }
-    if (!list_append(&ctx->passwords, &info, sizeof(info))) {
-        pgp_forget(&info, sizeof(info));
+    try {
+        ctx.passwords.emplace_back(info);
```

Comment:
why not `push_back`?

Reply:
@rrrooommmaaa In my understanding `emplace_back()` would directly construct object here, using the move constructor, while `push_back()` will first create temporary and then do copy/move. Is it wrong in this  case?

Reply:
@ni4 Just checking. We have lots of `push_back` calls throughout the app.
We even have `siginfos.push_back({})` which should probably be replaced with `emplace_back`
I'm not an expert in this field, but from what I read, `emplace_back` is useful when you pass a list of constructor arguments to it, so that it is constructed in place. But since `sinfo` is the object itself, I suspect "move construction" will happen in the same way as in `push_back` implementation. I'm just trying to figure out the general rule to use throughout the app.

Reply:
@rrrooommmaaa My opinion on this is like 'call emplace_back() when we move object/construct from the parameter list or push_back() when we copy', but it could be subject to change if there are some arguments I'm missing. Maybe @dewyatt has also some thoughts on this topic?

Reply:
I don't think there's an advantage to `emplace_back` in this case since we're inserting an existing object, not constructing and inserting. Since `emplace*` can have surprising behavior with implicit conversions, it may be best to stick with `push_back` except in cases where there is a clear advantage. (Effective Modern C++ has some guidance in Item 42).

Reply:
@dewyatt Thanks, let's stick to `push_back()` in such cases then.

Reply:
@rrrooommmaaa Changed to `push_back()` and re-pushed. Any other comments?